src/alloc.rs

```diff
@@ -133,14 +133,18 @@ unsafe impl core::alloc::Allocator for Allocator {
         // size as argument, and places a pointer to the allocation in an output argument. Note
         // that UEFI guarantees 8-byte alignment (i.e., `POOL_ALIGNMENT`). To support higher
         // alignments, see the `align_request() / align_block() / unalign_block()` helpers.
-        let mut ptr: *mut core::ffi::c_void = core::ptr::null_mut();
         let align = layout.align();
         let size = layout.size();
         let size_allocated = align_request(size, align);
 
-        if size > 0 {
+        let ptr_aligned = if size > 0 {
```

Comment:
I don't feel like this construct is more readable than the direct assignments. I understand that if-blocks can be nicely used as expressions, but I am not feeling like this is aiding readability here.